quickwit/quickwit-proto/src/metastore/mod.rs

```diff
@@ -17,7 +17,179 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
+use std::fmt;
+
+use quickwit_common::retry::Retryable;
+
+use crate::{IndexId, QueueId, ServiceError, ServiceErrorCode, SourceId, SplitId};
+
 include!("../codegen/quickwit/quickwit.metastore.rs");
 
 pub use metastore_service_client::MetastoreServiceClient;
 pub use metastore_service_server::{MetastoreService, MetastoreServiceServer};
+
+pub type MetastoreResult<T> = Result<T, MetastoreError>;
+
+/// Lists the object types stored and managed by the metastore.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum EntityKind {
+    /// A checkpoint delta.
+    CheckpointDelta {
+        /// Index ID.
+        index_id: IndexId,
+        /// Source Id.
+        source_id: SourceId,
+    },
+    /// An index.
+    Index {
+        /// Index ID.
+        index_id: IndexId,
```

Comment:
An index also has a UID. I guess in some cases we would want to refer to the UID.

Reply:
True. We kept displaying index IDs to the users when we introduced index UIDs, so I followed that convention.